<launch>

  <!-- common param -->
  <arg name="mode" default="0" /> <!-- uav side: 0; gcs side: 1 -->
  <arg name="log_output" default="screen" />

  <!-- aerial_robot_comm -->
  <arg name="port" default="/dev/ttyUSB0" />
  <arg name="baudrate" default="9600" />
  <arg name="debug_verbose" default="false" />
  <arg name="tx_debug" default="false" />
  <arg name="gcs_debug" default="false" />

  <!-- mavlink bridge -->
  <arg name="gcs_url" default="udp://@127.0.0.1:14550" />
  <arg name="tgt_system" default="1" />
  <arg name="tgt_component" default="240" />

  <!-- xbee ros interface -->
  <node  pkg="aerial_robot_comm" type="aerial_robot_comm_node" name="aerial_robot_comm" required="true" clear_params="true" output="$(arg log_output)">
    <param name="mode" value="$(arg mode)" />
    <param name="port" value="$(arg port)" />
    <param name="baudrate" value="$(arg baudrate)" />
    <param name="rssi_rate" value="0.25" />
    <param name="debug_verbose" value="$(arg debug_verbose)" />
    <param name="tx_debug" value="$(arg tx_debug)" />
    <param name="gcs_debug" value="$(arg gcs_debug)" />
  </node>

  <!-- mavlink bridge -->
  <group unless="$(arg tx_debug)"> 
<!--<group unless="1"> -->
    <node pkg="aerial_robot_comm" type="aerial_robot_comm_mavros_node" name="mavros" required="true" clear_params="true" output="$(arg log_output)">
      <param name="gcs_url" value="$(arg gcs_url)" if="$(arg mode)"/> <!-- gcs side -->
      <param name="target_system_id" value="$(arg tgt_system)" />
      <param name="target_component_id" value="$(arg tgt_component)" />

      <!-- load blacklist, config -->
      <rosparam command="load" file="$(find aerial_robot_comm)/config/uav_plugins.yaml" unless="$(arg mode)"/> <!-- uav side -->
      <rosparam command="load" file="$(find aerial_robot_comm)/config/gcs_plugins.yaml" if="$(arg mode)"/> <!-- gcs side -->

      <!-- mavros plugin (e.g ps3joy, odom) -->
      <param name="mode" value="$(arg mode)" />
    </node>
  </group>

  <!-- UAV side-->
  <group unless="$(arg mode)">
    <!-- TODO -->
    <!-- parser node for some ros message: bat_voltage, odometry(gps, height)-->
    <node pkg="aerial_robot_comm" type="aerial_robot_bridge" name="aerial_robot_bridge" required="true" clear_params="true" output="$(arg log_output)">
      <rosparam command="load" file="$(find aerial_robot_comm)/config/aerial_robot_plugins.yaml"/>
    </node>
  </group>

  <!-- GCS side-->
  <group if="$(arg mode)">
    <!-- TODO -->
    <!-- qground control start up: first wirte into CMakeLists.txt -->
  </group>

</launch>
